Hide non-numeric fields for Math calculations

shervElmi opened the pull request:
Fixes https://github.com/Strategy11/formidable-pro/issues/6087

When building a Math calculation, non-numeric fields (Name, Email, URL, Address, Text, etc.) are now automatically hidden from the field list. These fields don't make sense in mathematical operations, so hiding them reduces confusion and prevents errors.

When switching to Text calculation type, all fields are shown again since text calculations can combine any field values.

---

### Pro PR

https://github.com/Strategy11/formidable-pro/pull/6117

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Improvements**
  * Numeric calculation fields now display only compatible numeric field types when selecting fields for formulas.
  * Text-based calculations continue to support the existing field-selection behavior.
  * Excluded fields and additional fields are handled consistently when configuring calculation formulas.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

shervElmi committed:
Add helper method for non-numeric field types and fix trailing whitespace

shervElmi committed:
Exclude non-numeric field types from Math calculation field dropdowns

shervElmi committed:
Add event listener to refresh calc field list when field type changes

shervElmi committed:
Refactor calculation type check and remove unused event listener

Remove `isTextCalculationForBox` and `refreshCalcFieldListOnTypeChange` functions and replace with simpler `isMathCalcType` helper that directly checks the calc_type radio button value. Remove the change event listener for calc_type inputs as it's no longer needed.

shervElmi committed:
Add empty state message for Math calculations with no numeric fields

Show a message "This form has no numeric fields to insert into the calculation." and hide the search box when a Math calculation type has no available numeric fields to display. Add JSDoc comment for the popCalcFields function.

shervElmi committed:
Simplify Math calculation type check in getExcludeArray function

Refactor the Math calculation type check to use the `isMathCalcType` helper function instead of `isTextCalculationForBox`. Simplify the conditional logic and inline the JSON parsing. Add JSDoc comment for the getExcludeArray function.

shervElmi committed:
Optimize calculation type check by caching isMathCalcType result

Cache the result of `isMathCalcType()` call in `popCalcFields()` to avoid redundant checks. Pass the cached value to `getExcludeArray()` function. Add optional chaining to `classList.contains()` check in `isMathCalcType()` for safer property access.

shervElmi committed:
Add top padding to empty state message for Math calculations with no numeric fields

shervElmi committed:
Merge remote-tracking branch 'origin/HEAD' into hide-non-numeric-fields-math-calc

shervElmi committed:
Add auto generated files

lauramekaj1 reviewed (approved):
@shervElmi I tested it and it's working as expected. Thank you!

shervElmi committed:
Move non_numeric_field_types method to FrmProAppHelper class

Relocate the `non_numeric_field_types()` static method from `FrmAppHelper` to `FrmFieldsHelper` where field-type-related utility methods belong. Remove the method from FrmAppHelper as it's more appropriately scoped to field operations.

shervElmi committed:
Refactor calculation field filtering to use allowlist instead of excludelist for numeric types

Replace the exclude-based approach for filtering non-numeric fields in Math calculations with an allowlist approach. Move numeric field type filtering from `getExcludeArray()` to the main field loop in `popCalcFields()`. Add `getNumericFieldTypes()` helper function to retrieve allowed numeric types from `data-numeric-types` attribute. Simplify `getExcludeArray()` by removing the `isMathCalc` parameter

shervElmi committed:
Add formidable_admin.js file

Crabcyborg commented on classes/helpers/FrmAppHelper.php:
```diff
@@ -5071,6 +5071,31 @@ public static function get_doc_url( $path, $campaign, $add_kb_base = true ) {
+	 *
+	 * @return array
+	 */
+	public static function non_numeric_field_types() {
```
@shervElmi Your list includes `select`, which should cover dropdowns.

We might need to use `dropdown` instead though. We use both strings to refer to dropdowns, as `select` is unsafe to actually send in POST data (I believe modsecurity would block all requests that include the word `'select'` in the `$_POST` data, flagging them incorrectly as raw SQL).

lauramekaj1 reviewed (approved):
@shervElmi I tested it and it's working as expected. Thank you!

Crabcyborg committed:
Merge branch 'master' into hide-non-numeric-fields-math-calc

Crabcyborg committed:
Prefer a filter

Crabcyborg reviewed (approved):
I removed the Pro logic from this branch in favour of using a new hook.

This should be good to go now.

🚀
